### PR TITLE
[CI] Automate AUR Publishing

### PR DESCRIPTION
**Pull request by flavioislima**
Github Workflow to update AUR after a new release.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: .github/workflows/release_aur.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          source=("$url/releases/download/$latest_version/$_filename")
+          md5sums=($(md5sum heroic.pacman | cut -d' ' -f1))
+
+          cat <<EOF > PKGBUILD
+          # Maintainer: flaviofearn <flavioislima@gmail.com>
+
+          pkgname=heroic-games-launcher-bin
+          pkgver=$pkgver
+          pkgrel=$pkgrel
+          pkgdesc="An open source Web3 gaming platform."
```
Comment (CommandMC) [inline .github/workflows/release_aur.yml:45]: That doesn't look right

**Review by CommandMC — COMMENTED** [review]
A user on the AUR suggested [this PKGBUILD](https://github.com/FabioLolix/AURFIX/blob/main/h/heroic-games-launcher-bin/PKGBUILD) instead, maybe we can build off that here?

File: .github/workflows/release_aur.yml (hunk 1 above)
Comment (flavioislima) [reply .github/workflows/release_aur.yml:45]: good catch, copy pasted from hyperplay 🤦🏽

Comment (flavioislima) [reply .github/workflows/release_aur.yml]: fixed

**Comment by flavioislima** [comment]
> A user on the AUR suggested [this PKGBUILD](https://github.com/FabioLolix/AURFIX/blob/main/h/heroic-games-launcher-bin/PKGBUILD) instead, maybe we can build off that here?

doesn't look right to me, why Heroic would depend on gtk3? I cannot see any other difference besides the dependencies., do you?

**Comment by Etaash-mathamsetty** [comment]
> > A user on the AUR suggested [this PKGBUILD](https://github.com/FabioLolix/AURFIX/blob/main/h/heroic-games-launcher-bin/PKGBUILD?rgh-link-date=2023-10-06T19%3A20%3A34Z) instead, maybe we can build off that here?
> 
> doesn't look right to me, why Heroic would depend on gtk3? I cannot see any other difference besides the dependencies., do you?

electron depends on gtk3 to my knowledge

**Comment by flavioislima** [comment]
> > > A user on the AUR suggested [this PKGBUILD](https://github.com/FabioLolix/AURFIX/blob/main/h/heroic-games-launcher-bin/PKGBUILD?rgh-link-date=2023-10-06T19%3A20%3A34Z) instead, maybe we can build off that here?
> > 
> > 
> > doesn't look right to me, why Heroic would depend on gtk3? I cannot see any other difference besides the dependencies., do you?
> 
> electron depends on gtk3 snip from ldd output:
> 
> ```
> libgtk-3.so.0 => /usr/lib/libgtk-3.so.0 (0x00007f5b4d600000)
> ```

but we are not using electron from the system, so I dont see the point. The build from pacman we use already contains all the dependencies.

**Comment by Etaash-mathamsetty** [comment]
> > > > A user on the AUR suggested [this PKGBUILD](https://github.com/FabioLolix/AURFIX/blob/main/h/heroic-games-launcher-bin/PKGBUILD?rgh-link-date=2023-10-07T15%3A23%3A50Z) instead, maybe we can build off that here?
> > > 
> > > 
> > > doesn't look right to me, why Heroic would depend on gtk3? I cannot see any other difference besides the dependencies., do you?
> > 
> > 
> > electron depends on gtk3 snip from ldd output:
> > ```
> > libgtk-3.so.0 => /usr/lib/libgtk-3.so.0 (0x00007f5b4d600000)
> > ```
> 
> but we are not using electron from the system, so I dont see the point. The build from pacman we use already contains all the dependencies.

not true, the current heroic pacman package doesn't ship gtk3

**Comment by CommandMC** [comment]
> I cannot see any other difference besides the dependencies., do you?

You can see the full list of changes in [this commit](https://github.com/FabioLolix/AURFIX/commit/9d7f1ff8405e5594673ecb3d9404d4f02c945e85)
They've...
- set `url` to our official website (which, according to the AW, should be used if available)
- corrected `provides` and `conflicts` (we provide `heroic-games-launcher` and also conflict with all other packages providing it)
- used `sha256sums` instead of `md5sums`, which is generally preferred
- changed the dependencies as you mentioned

**Comment by flavioislima** [comment]
> > > > > A user on the AUR suggested [this PKGBUILD](https://github.com/FabioLolix/AURFIX/blob/main/h/heroic-games-launcher-bin/PKGBUILD?rgh-link-date=2023-10-07T15%3A23%3A50Z) instead, maybe we can build off that here?
> > > > 
> > > > 
> > > > doesn't look right to me, why Heroic would depend on gtk3? I cannot see any other difference besides the dependencies., do you?
> > > 
> > > 
> > > electron depends on gtk3 snip from ldd output:
> > > ```
> > > libgtk-3.so.0 => /usr/lib/libgtk-3.so.0 (0x00007f5b4d600000)
> > > ```
> > 
> > 
> > but we are not using electron from the system, so I dont see the point. The build from pacman we use already contains all the dependencies.
> 
> not true, the current heroic pacman package doesn't ship gtk3, or a majority of the libraries that electron depends on

any reason why it is not breaking? or did it break to someone already because of these dependencies?
I am worried that we might not need gtk3 and we will force people to install it when installing heroic, it is being almost 3 years with this AUR repo and we never had these deps there.

**Comment by Etaash-mathamsetty** [comment]
> > > > > > A user on the AUR suggested [this PKGBUILD](https://github.com/FabioLolix/AURFIX/blob/main/h/heroic-games-launcher-bin/PKGBUILD?rgh-link-date=2023-10-18T11%3A58%3A16Z) instead, maybe we can build off that here?
> > > > > 
> > > > > 
> > > > > doesn't look right to me, why Heroic would depend on gtk3? I cannot see any other difference besides the dependencies., do you?
> > > > 
> > > > 
> > > > electron depends on gtk3 snip from ldd output:
> > > > ```
> > > > libgtk-3.so.0 => /usr/lib/libgtk-3.so.0 (0x00007f5b4d600000)
> > > > ```
> > > 
> > > 
> > > but we are not using electron from the system, so I dont see the point. The build from pacman we use already contains all the dependencies.
> > 
> > 
> > not true, the current heroic pacman package doesn't ship gtk3, or a majority of the libraries that electron depends on
> 
> any reason why it is not breaking? or did it break to someone already because of these dependencies? I am worried that we might not need gtk3 and we will force people to install it when installing heroic, it is being almost 3 years with this AUR repo and we never had these deps there.

oh if no one has complained then never mind, it might work just fine

**Comment by flavioislima** [comment]
> 
> oh if no one has complained then never mind, it might work just fine

Not sure if you are being ironic or not but that is actually a valid point.

**Pull request merged**